lib/common_test/src/test_server_ctrl.erl

```diff
@@ -3616,28 +3620,32 @@ handle_test_case_io_and_status() ->
 			     (_) ->
 				  ok
 		  end, Cases),
-	    Result
+	    {{Ok, Skip, Fail}, Time}
     end.
 
 %% Handle cases (without Ref) that belong to the top parallel group (i.e. when Refs = [])
-handle_io_and_exit_loop([], [{undefined,CurrPid,CaseNum,Mod,Func}|Ps] = Cases, Ok,Skip,Fail) ->
+handle_io_and_exit_loop([], [{undefined,CurrPid,CaseNum,Mod,Func}|Ps] = Cases, Ok,Skip,Fail, Time) ->
     %% retrieve the start message for the current io session (= testcase)
     receive
 	{started,_,CurrPid,CaseNum,Mod,Func} ->
-	    {Ok1,Skip1,Fail1} =
+	    {Ok1,Skip1,Fail1,Time1} =
 		case handle_io_and_exits(self(), CurrPid, CaseNum, Mod, Func, Cases) of
-		    {ok,MF} -> {[MF|Ok],Skip,Fail};
-		    {skipped,MF} -> {Ok,[MF|Skip],Fail};
-		    {failed,MF} -> {Ok,Skip,[MF|Fail]}
+		    {ok,MF, RTime} -> {[MF|Ok],Skip,Fail, RTime};
+		    {skipped,MF, 0} -> {Ok,[MF|Skip],Fail, 0};
+		    {failed,MF, RTime} -> {Ok,Skip,[MF|Fail], RTime}
 		end,
-	    handle_io_and_exit_loop([], Ps, Ok1,Skip1,Fail1)
+		GroupExecutionTime = case Time1 > Time of
+                                true -> Time1;
+							    _ -> Time
+		                    end,
```

Comment:
general comment, please check whole PR.

space after `,` character - in added or modified code, please try to not to mix convention in same tuple or in argument list to a function -  to not make reading experience worse.